packages/server/src/mobs/mob.ts

```diff
@@ -363,7 +371,34 @@ export class Mob {
     }
   }
 
-    changeSpeed(amount: number) {
+  // Addition: These things happen when the character is in sleep state (healing + energy)
+  sleep() {
+    this.needs.changeNeed('max_energy', 25);
+    this.needs.changeNeed('energy', 25);
+    this.changeHealth(10);
+  }
+
+  // Addition: Implements sleep if it recognizes that mob is asleep
+  checkForSleep() {
+    const currentTick = gameWorld.currentDate().global_tick; // Use the game world time and not real time
+
+    if (currentTick - this.lastMoveTick >= this.sleepDuration) {
+      this.sleep();  // Sleep if condition is met
+      this.lastMoveTick = currentTick; // Reset the last move tick
+      DB.prepare(
+        `
+              UPDATE mobs
+              SET  lastMoveTick = :lastMoveTick
+              WHERE id = :id
+          `
+      ).run({
+        id: this.id,
+        lastMoveTick: this.lastMoveTick,
+      });
+    }
+  }
+  
```

Comment:
I'm a bit confused as to why this needs to be added when this "sleep plan" exists: https://github.com/sloalchemist/potions/blob/fdb1927886396ac588682941dd019dc097f7f159/packages/server/src/mobs/plans/sleep.ts#L8

Can you explain how this is different and how it fits in with the existing system?